index.bs

```diff
@@ -1101,31 +1127,116 @@ strong map from handle ids to their corresponding objects.
 OwnershipModel = "root" / "none";
 </pre>
 
-<dfn>RemoteReference</dfn> represents a remote reference to an existing ECMAScript
-object in [=handle object map=] in the given [=Realm=].
+<dfn>RemoteReference</dfn> is either a <code>RemoteObjectReference</code>
+representing a remote reference to an existing ECMAScript object in
+[=handle object map=] in the given [=Realm=], or is a <code>SharedReference</code>
+representing a reference to a [=node=].
 
 <pre class="cddl remote-cddl local-cddl">
 Handle = text;
+SharedId = text;
+</pre>
+
+<pre class="cddl remote-cddl">
+<!-- This is specifically ordered in the order in which matches need to be -->
+<!-- evaluated, since the definitions are overlapping -->
+RemoteReference = (
+  SharedReference //
+  RemoteObjectReference
+)
 
-RemoteReference = {
+SharedReference = {
+   sharedId: SharedId
+   <!-- Ensure that if we have a handle, it at least has the correct type -->
+   ?handle: Handle,
```

Comment:
It means that you can't write:
```
{
   sharedId: "foo",
   handle: ["some other type"]
}
```
which otherwise would match due to `Extensible`. Given that the proposed use case is "we want to allow people to directly send JSON data from responses" it makes sense to me to ensure that we can't extend this in a way that doesn't make sense (but honestly I still think we should care less about this use case and prefer nomnative typing over structural typing. However I don't seem to have much support for that position).